source/agora/network/Manager.d

```diff
@@ -194,6 +194,7 @@ public class NetworkManager
                 }
 
                 log.info("Found new Validator: {} (key: {})", address, key);
+                this.outer.banman.whitelist(address);
```

Comment:
Well... This just checks if the node implements a certain API. It would be trivial to write a node that exploits this not to get banned, no ?

Reply:
Yep, we could whitelist them as long as they are enrolled maybe? A not enrolled validator is no different than a FullNode anyways.

Reply:
Yes we should. But obviously this means connecting the `BanManager` to the `Ledger`.

Reply:
How about `onAcceptedBlock`to get a list of changes in validator set instead of just a bool?

Reply:
@Geod24, whitelisting enrolled validators are not a perfect solution as well. Anyone could impersonate a Validator at network level right? Maybe discovery handshake should include an authentication step.

Reply:
> Maybe discovery handshake should include an authentication step.

You're taking the words out of my mouth :)

Reply:
> How about `onAcceptedBlock` to get a list of changes in validator set instead of just a bool?

Sounds about right.